cpu/atmega_common/periph/gpio.c

```diff
@@ -127,6 +129,8 @@ static inline int8_t _int_num(gpio_t pin)
 
 int gpio_init(gpio_t pin, gpio_mode_t mode)
 {
+    GPIO_INTERCEPT_INIT(pin, mode);
```

Comment:
Having to add code to every single implementation needs to be avoided. Please find another way.

Reply:
Fixed. The intercepts are now implemented by splitting the GPIO functions into a low-level version.